subsys/net/lib/sockets/Kconfig

```diff
@@ -105,6 +105,7 @@ config NET_SOCKETS_SERVICE_THREAD_PRIO
 config NET_SOCKETS_SERVICE_STACK_SIZE
 	int "Stack size for the thread handling socket services"
 	default 2400 if NET_DHCPV4_SERVER
+	default 1400 if DNS_RESOLVER
 	default 1400 if MDNS_RESPONDER
```

Comment:
Please change this to
```
default 1400 if MDNS_RESPONDER || DNS_RESOLVER
```

Reply:
This is complicated if DNS_MAX_NAME_SIZE is increased to an arbitrary value. 1400 may not be enough.
I would argue that DNS_MAX_NAME_SIZE should be 255 to be in line with [RFC1035](https://www.ietf.org/rfc/rfc1035.txt). I understand reducing memory usage for many cases in which domains are shorter, but I can see where this will be very confusing trying to get the right stack size for the net sockets service when using DNS and a dynamic DNS max length.

Perhaps the default should should be 255 and the stack size should be large enough to accommodate that, but users could adjust it to be a smaller value to reduce memory usage.

Any thoughts?

Reply:
Actually, looking into it more, [`DNS_RESOLVER_MAX_QUERY_LEN`](https://github.com/zephyrproject-rtos/zephyr/blob/e0dbba5bd3ff18103fa3adb9c6efe2e44fcf278b/subsys/net/lib/dns/Kconfig#L64) seems to already be a Kconfig for this value, if I'm not mistaken. It's default is 255. I'm not sure why there is a redundant `DNS_MAX_NAME_SIZE` that is only 20 bytes.

Reply:
We can certainly have a dedicated line for stack size if the 1400 is not enough.

But lets not increase the default value to such a high value, you are the first person that has reported this problem. So if it is possible to increase the max name len by the user via kconfig, then we are not introducing any changes how the system behaves by default.

Reply:
> I'm not sure why there is a redundant `DNS_MAX_NAME_SIZE` that is only 20 bytes.

The other one is used in query, the other is where the response is stored. This has worked well so far, but if we have a Kconfig for setting `DNS_MAX_NAME_SIZE`, then user is certainly able to set those to same value.

Reply:
Okay, so it seems the hawkbit subsystem has been misusing `DNS_MAX_NAME_SIZE`, as it is using it to size the string for the query. I propose replacing this with `DNS_RESOLVER_MAX_QUERY_LEN` where appropriate in the hawkbit code, and returning `DNS_MAX_NAME_SIZE` and the stack size configs to their original values.

Does this sound correct?

Reply:
Yes, sounds about right.

Reply:
This has been split to #89497